### ISSUE TITLE
13.0.3 - On Mobile: Bad utilization of space for Folder/File Names on Website

### ISSUE THREAD

**Issue by johnyb0y** [nextcloud/server#9817]
As requested by @MorrisJobke - New issue request, since it's still happening on 13.0.3 

**See original for details:** https://github.com/nextcloud/server/issues/8940
**Related PR:** https://github.com/nextcloud/server/pull/9248

I'd be happy to run some tests if you need additional details.
Thanks.

**Comment by juliushaertl** [comment]
@johnyb0y Please provide some details about where exactly is this happening, like the public shared file list or the regular file list of a user? Some details on your browser would also be great.

For me the fix works quite fine inside the files app as well as on public shared folders with Chrome/Firefox on Android.

**Comment by johnyb0y** [comment]
iPhone 7, latest iOS Version, using Safari (though afaik every iOS Browser uses the same Webkit engine):

Using the same folder with subfolders and files:

**Registered user:** Everything fine!
**Public Shared link:** Still happening.

Thanks for looking into it.

**Comment by MorrisJobke** [comment]
@johnyb0y Works here just fine on a 13.0.3 instance:

![img_c416a9acfec9-1](https://user-images.githubusercontent.com/245432/41215684-cb32dfb0-6d51-11e8-9d23-d035792dabde.jpeg)

**Comment by johnyb0y** [comment]
![photo_2018-06-11_08-37-14](https://user-images.githubusercontent.com/12081528/41216025-0d926fb4-6d53-11e8-95ab-bb2620430285.jpg)

**Nextcloud Version:**
![2018-06-11 08_39_14-window](https://user-images.githubusercontent.com/12081528/41216048-17fb5240-6d53-11e8-8c2e-7a0e493b6eb3.png)

I use the manual update method with occ and not the Webupdater, but i guess that shouldn't matter.

**Comment by MorrisJobke** [comment]
One thing I could imagine: the caching went wrong. To verify this, could you remove the files and folders inside `DATADIR/appdata_INSTANCEID/css` and then run `occ files:scan-app-data` to rescan that directory. Then on the next request all assets are re-generated. Does this fix your problem?

**Comment by johnyb0y** [comment]
I'm gonna check this and report back when I get home from work.

**Comment by juliushaertl** [comment]
@MorrisJobke @johnyb0y We also have included a cache cleanup in the repair occ command, so running `occ maintenance:repair` should do the same. :wink:

**Comment by johnyb0y** [comment]
@juliushaertl @MorrisJobke 
Sooo, heres what i did:


1. As user root: `rm -r /var/www/html/nextcloud/data/appdata_xxxxxxxxxxxxx/css/*`

2. `sudo -u www-data /var/www/html/nextcloud/./occ files:scan-app-data`

```
Scanning AppData for files

+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 4310    | 34414 | 00:00:17     |
+---------+-------+--------------+
```

3. `sudo service apache2 restart`

**Result:** No Change.


Additionally: `sudo -u www-data /var/www/html/nextcloud/./occ maintenance:repair`

```
 - Repair MySQL collation
     - All tables already have the correct collation -> nothing to do
 - Repair mime types
 - Clean tags and favorites
     - 0 tags of deleted users have been removed.
     - 0 tags for delete files have been removed.
     - 0 tag entries for deleted tags have been removed.
     - 0 tags with no entries have been removed.
 - Repair invalid shares
 - Remove shares of a users root folder
 - Move .step file of updater to backup location
 - Add move avatar background job
     - Repair step already executed
 - Add preview cleanup background jobs
 - Fix potential broken mount points
     - No mounts updated
 - Repair language codes
 - Install new core bundle components
 - Repair invalid paths in file cache
 - Rename folder with user specific keys
 - Add log rotate job
 - Clear frontend caches
     - Image cache cleared
     - SCSS cache cleared
     - JS cache cleared
 - Fix component of birthday calendars
     - 1 birthday calendars updated.
 - Fix broken values of calendar objects
    0 [>---------------------------]
 - Registering building of calendar search index as background job
     - Repair step already executed
 - Fix the share type of guest shares when migrating from ownCloud
 - Copy the share password into the dedicated column
 - Move theming files to AppData storage
     - No theming logo image to migrate
     - No theming background image to migrate
```

**Result:** No Change.

I also cleared Cache on my iOS device, just to be sure.

**Comment by MorrisJobke** [comment]
@johnyb0y As we can't reproduce, do you mind to send us a share link so we can have a look at it on our own devices?

**Comment by johnyb0y** [comment]
Of course. 
Because I don’t want to expose my Nextcloud publicly i‘d like to send the link via email. Okay ?

**Comment by juliushaertl** [comment]
@johnyb0y Sure, just send me the link to the address mentioned in my GitHub profile, then i'll have a look.

**Comment by johnyb0y** [comment]
Sent both of you a mail!

**Comment by juliushaertl** [comment]
Looks like the css that is served is actually outdated, but I have no clue why this could happen if the appdata folder has been deleted, since the SCSSCacher would then regenerate all files.

From the test link:
![image](https://user-images.githubusercontent.com/3404133/41233510-effa71f4-6d89-11e8-9783-8e79a925a520.png)

@johnyb0y Can you check if the file `apps/files/css/files.scss` is identical to the one from the release archive?

**Comment by johnyb0y** [comment]
I may have an idea. Maybe while installing manually I accidentally merged the old apps folder into my new ownCloud directory. 
I don’t think this happened, but I can’t rule it out :-) 
I will check the css you mentioned this evening.

**Comment by johnyb0y** [comment]
@juliushaertl @MorrisJobke 

Things I tested:

1) @juliushaertl asked me to take a look at apps/files/css/files.scss and compare it to the one in the 13.0.3 zip archive:

**Result:** They are identical.


2) Too rule out me making some mistake with the last manual update to 13.0.3 I removed everything in my `apps` folder (except `bruteforcesetttings` and `previewgenerator` since they are not part of the default nextcloud apps) , and copied everything from the `apps` folder in the 13.0.3 zip archive.

I then did an `occ upgrade, occ occ files:scan-app-data and occ maintenance:repair`

I also deleted the css cache again like @MorrisJobke explained earlier.

**Result:** No Change.

3) Meanwhile you have just released **13.0.4**, so I used the Web-Updater to update this time.
I use a custom .htaccess which i add after every update, but other than that this a pretty stock nextcloud deployment.

Again, deleted the css cache.

## **Result:** It works!

I have no idea what happened here. Something still had the old scss cached.
Anyway, big thanks for your help 👍
